bolt-simple-taproot.md

```diff
@@ -0,0 +1,1325 @@
+
+A `musig2` public nonce is technically the concatenation of two public keys,
+each representing the EC-point corresponding to its secret integer, thus
+resulting in a 66-byte value:
+
+```
+point_1 || point_2
+```
+
+For nonce generation, we refer to the `NonceGen` algorithm defined in the
+`musig2` BIP. We only specify the set of required arguments, though a signer
+can also opt to specify the additional argument, in order to strengthen their
+nonce.
+
+##### Counter Based Nonce Generation & JIT Nonce Strengthening 
+
+Each new commitment state requires fresh nonces from both parties, as each
+commitment state is signed using a fresh musig2 session. Nonces sent by either
+party in the `commitment_signed` message can be generated Just-In-Time (JIT),
+thus requiring no additional state. These nonces can also be further
+strengthened by mixing in the commitment sighash information into the nonce
+generation algorithm. 
+
+Upon channel creation, and subsequent re-establishment, a fresh set of
+verification nonces are exchanged by each party. This is the nonce the party
+will use to _verify_ a new incoming commitment state sent by the remote party.
+In order to force close a channel, the holder of a verification nonce must use
+that same nonce to counter sign the commitment transaction with the other half
+of the musig2 partial signature. Rather than force an implementation to retain
+additional signing state (the verification nonce) to avoid holding as "hot"
+(fully broadcast able) commitment on disk, we instead recommend a counter based
+approach.
+
+In this scenario are nonce generated via a counter is deemed to be safe as:
+
+  1. Each commitment state only has a single signature exchanged to counter
+     sign the commitment transaction.
+
+  2. Each commitment state has a unique number which is currently encoded as a
+     48-bit integer across the sequence and lock time of the commitment
+     transaction. 
+
+  3. The shachain scheme is used today to generate a fresh nonce-like value for
+     the revocation scheme of today's penalty based channels.
+
+  4. A verification nonce is only used to _sign_ a local party's commitment
+     transaction when they go to broadcast a force close transaction.
+
+  5. The remote party never sees the unaggregated partial signature the local
+     party will use to broadcast, only the aggregated signature is seen on chain.
+
+  6. As a result of all the above, a verification nonce will never be used to
+     sign the exact same commitment transaction, thereby avoiding nonce reuse.
+
+As a result, we recommend the following counter based scheme to generate
+verification nonce. With this scheme, signing can remain stateless, as given
+the commitment height/number of the state to sign, and the original shachain
+root, the verification nonce sent by that state can be deterministically
+reproduced:  
+
+  1. Given the shachain root used to generate revocation pre-images,
```

Comment:
has any thought been given on how this handles/could be extended to handle splicing? 

you're going to possibly be signing multiple times at a given height, and sharing nonces before knowing the final txid/state of the commit tx.

Without giving much thought the splices could be given a counter and incorporated to the shachain scheme.

cc @ddustin